src/js/controllers/walletHome.js

```diff
@@ -335,69 +335,70 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
 
       $scope.sign = function(txp) {
         var fc = profileService.focusedClient;
+        $scope.error = null;
 
         if (!fc.canSign() && !fc.isPrivKeyExternal())
           return;
 
-        if (fc.isPrivKeyEncrypted()) {
-          profileService.unlockFC(function(err) {
-            if (err) {
-              $scope.error = bwsError.msg(err);
-              return;
-            }
-            return $scope.sign(txp);
-          });
-          return;
-        };
-
-        self._setOngoingForSigning();
         $scope.loading = true;
-        $scope.error = null;
         $timeout(function() {
+
           requestTouchid(function(err) {
             if (err) {
-              self.setOngoingProcess();
               $scope.loading = false;
-              profileService.lockFC();
               $scope.error = err;
-              $scope.$digest();
+              $timeout(function() {
+                $scope.$digest();
+              });
               return;
             }
 
-            profileService.signTxProposal(txp, function(err, txpsi) {
-              self.setOngoingProcess();
+            profileService.unlockFC(function(err) {
```

Comment:
we should not call unlock if the wallet is not locked. that will ask a password.